src/core/qgsdatasourceuri.h

```diff
@@ -318,6 +319,12 @@ class CORE_EXPORT QgsDataSourceUri
      */
     void setGeometryColumn( const QString &geometryColumn );
 
+    /**
+     * Returns parameter keys used in the uri: specialized ones ("table", "schema", etc.) or generic parameters.
+     * \since QGIS 3.26
+     */
+    const QSet<QString> &parameterKeys() const;
```

Comment:
```suggestion
    QSet<QString> parameterKeys() const;
```

(since QSet is implicitly shared, and it avoids any risk of lifetime issues wrt python code)

Reply:
done